bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/FileSystemUpgrade.java

```diff
@@ -99,7 +100,13 @@ public boolean accept(File dir, String name) {
     private static List<File> getAllDirectories(ServerConfiguration conf) {
         List<File> dirs = new ArrayList<>();
         dirs.addAll(Lists.newArrayList(conf.getJournalDirs()));
-        Collections.addAll(dirs, conf.getLedgerDirs());
+        final File[] ledgerDirs = conf.getLedgerDirs();
+        final File[] indexDirs = conf.getIndexDirs();
+        if (indexDirs != null
+                && !Arrays.asList(indexDirs).equals(Arrays.asList(ledgerDirs))) {
```

Comment:
`Arrays.asList(indexDirs).equals(Arrays.asList(ledgerDirs))` 
How to judge the correctness of this sentence? I think you need to add two more examples to cover this code

Reply:
@StevenLuMT You are right! I made a mistake here, I updated the method and added the test to cover it, please help to review, thank you!

Reply:
it is not right
ledgerDirs and indexDirs are one-to-one mappings
If this state occurs, it feels that there will be an error in judgment, which will lead to more additions?

ledgerDirs=/t1,/t2,/t3
indexDirs=/t1,/i1,/i3 

> @StevenLuMT You are right! I made a mistake here, I updated the method and added the test to cover it, please help to review, thank you!

Reply:
@StevenLuMT I have considered the 1-to-1 situation you mentioned, but I think that in this tool, it does not care whether it is 1-to-1, but only focuses on whether the number of ledgerDirs and indexDirs and whether the absolute paths are the same.

Reply:
If you are worried about the existence of duplicate directories, resulting in multiple executions, then we can use the Set collection, but the original implementation did not consider whether JournalDirs and LedgerDirs are duplicated? Also, 1 vs 1 doesn't make a difference here.

Reply:
ok, LGTM, let Enrico or others review it again @wenbingshen 

> If you are worried about the existence of duplicate directories, resulting in multiple executions, then we can use the Set collection, but the original implementation did not consider whether JournalDirs and LedgerDirs are duplicated? Also, 1 vs 1 doesn't make a difference here.

Reply:
@StevenLuMT Thanks. :)